server/find_knn.py

```diff
@@ -16,6 +16,7 @@
 paper_models = create_paper_models(N_NEIGHBORS)
 
 
+@timeout(seconds=240)
```

Comment:
Is this suppose to be 4 minutes? I was under the impression it was five minutes. I'm okay with either way.

Reply:
Good question. There are actually two timeout values. If no results is found in 4 minutes, the response will be an error message. On the other hand, the WSGI server (gunicorn) has another timeout, which is set to 5 minutes. For other errors that are not caught by the first timeout, the app server will restart.